lib/base/base_controller.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:async';
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+mixin BaseController {
+
+  final viewState = ValueNotifier<Either<Failure, Success>>(Right(UIState.idle));
```

Comment:
Ideally, we keep this because we only need `single source of truth`

Reply:
https://dev.to/aryclenio/unidirectional-and-bidirectional-data-flow-the-ultimate-front-end-interview-questions-guide-pt-1-5cnc

Reply:
But I think it is still not good for this project, because if we do this, some unrelated UI still be rebuild with some change of other `state`